File: scripts/replace.rb
#!/usr/bin/ruby -Ke

lang = ARGV.shift
common = ARGV.shift
template = ARGV.shift

t = open(template)
t.each{ |line|
  l = open(lang)
  l.each{ |word|
    word.chop!
    next if word == ""
    line.gsub!(/images\/#{word}/, "images/{$lang}/" + word)    
  }
  l.close;

  c = open(common)
  c.each{ |word|
    word.chop!
    next if word == ""
    line.gsub!(/images\/#{word}/, "images/common/" + word)    
  }
  c.close;
  
  print line;
}
t.close;

File: scripts/replace.pl
open(STDIN, "-");
while(<STDIN>){
    $line = $_;
    open(FILE, "<".$ARGV[0]);
    while(<FILE>){
	$word = $_;
	chop($word);
	$line =~ s/images\/$word/images\/\$\{lang\}\/${word}/;
	break;
    }
    close(FILE);

    open(FILE2, "<".$ARGV[1]);
    while(<FILE2>){
	$word2 = $_;
	chop($word2);
	$line =~ s/images\/${word2}/images\/common\/${word2}/;
	break;
    }
    close(FILE2);
   
    print $line;
}


close(STDIN);
